intellij-plugin-structure/structure-intellij/src/main/java/com/jetbrains/plugin/structure/intellij/plugin/PluginCreator.kt

```diff
@@ -734,6 +752,31 @@ internal class PluginCreator private constructor(
     }
   }
 
+  /**
+   * Checks if the _bean_ has any core plugin bean errors in the plugin descriptor.
+   *
+   * Such core errors are field validation errors (missing fields, empty values).
+   * If any such core error is found, an instance of invalid plugin with the most
+   * basic information is created and filtered by [problem resolver](problemResolver).
+   *
+   * This enables problem resolver to intentionally ignore core plugin bean errors
+   * and reclassify these errors into warnings or even ignore them.
+   *
+   * The usual example is an internal JetBrains plugin that might violate rules for
+   * allowed field values in the plugin descriptor.
+   *
+   * @see validatePluginBean
+   */
+  private fun hasPluginBeanErrors(bean: PluginBean, document: Document): Boolean {
+    if (problems.isEmpty()) {
+      return false
+    }
+    val invalidPlugin = newInvalidPlugin(bean, document)
+    return problemResolver.classify(invalidPlugin, problems).any {
```

Comment:
`PluginCreator` is actually using a two-phase validation.

1. In the first phase, the basic plugin descriptor properties are validated: mandatory fields, field names, plugin ID rules.
2. In the second phase, the complex plugin descriptor properties are checked: dependencies, modules etc.

When the first phase emits any problems, the second phase will not occur.

Originally, the Problem Resolver was not able to do anything in the first phase, as the plugin instance was not even created (there were only raw XML data and a `PluginBean` which is essentially a type-safe mirror of the XML).

Hence this _invalid plugin_. If the first phase fails, at least an invalid plugin is created. Then such plugin can passed down to the problem resolver which might remap even first phase plugin problems into non-problems.

For now, only the minimalistic info is passed down in the invalid plugin (ID, vendor, name), so the JetBrains plugin problem resolver can decide which JetBrains-specific issues will be remapped.

This might sound convoluted, but it can cover complex scenarios with multiple sets of rules.